src/monster.cpp

```diff
@@ -491,8 +491,9 @@ Creature::Attitude monster::attitude_to( const Creature &other ) const
             // so if both monsters are friendly (towards the player), they are friendly towards
             // each other.
             return A_FRIENDLY;
-        } else if( friendly == 0 && m->friendly == 0 ) {
-            // For now monsters are neutral (not hostile!) to other monsters.
+        } else if( friendly == 0 && m->friendly == 0 &&
+                   *type->species.begin() == *m->type->species.begin() ) {
```

Comment:
Instead of `*type->species.begin()` you should make a monster of mtype function that returns this data. It can later be changed into a more sophisticated faction system (your dog is on your side/faction, the npcs dog is on their side/faction, the wild dog is on a third faction).

Repeating `type->species.begin()` everywhere makes this more complicated. Seeing code like `this->getFaction() == other->getFaction()` is self documenting (comparing factions for obvious reasons), comparing the species is less obvious (why only the first one?).